src/scripts/main.js

```diff
@@ -1,3 +1,76 @@
 'use strict';
 
-// write code here
+document.addEventListener('DOMContentLoaded', () => {
+  const appendRowButton = document.querySelector('.append-row');
+  const removeRowButton = document.querySelector('.remove-row');
+  const appendColumnButton = document.querySelector('.append-column');
+  const removeColumnButton = document.querySelector('.remove-column');
+
+  const table = document.querySelector('.field');
+
+  const MAX_ROWS = 10;
+  const MIN_ROWS = 2;
+  const MAX_COLUMNS = 10;
+  const MIN_COLUMNS = 2;
+
+  const updateButtonsState = () => {
+    const rowCount = table.rows.length;
+    const columnCount = rowCount > 0 ? table.rows[0].cells.length : 0;
```

Comment:
The code assumes that the table will always have at least one row, which might not be the case. Consider adding a check to ensure that `table.rows[0]` is accessed only if `rowCount > 0` to avoid potential errors when the table is empty.